tests/test_integration.py

```diff
@@ -0,0 +1,33 @@
+from pathlib import Path
+import subprocess
+import json
+import pytest
+from typing import Dict, List, Tuple
+
+this_dir = Path(__file__).parent.resolve()
+fixtures_dir = this_dir / 'fixtures'
+exe_path = this_dir / '..' / 'build' / 'submitty_count_ts'
+
+def execute_command(file: str, command: str) -> int:
+    args = [exe_path]
+    args.extend((f"-l {file.split('.')[1]} {command} {str(this_dir / 'fixtures' / file)}").split(" "))
+    result = subprocess.run(args, capture_output=True)
+    output = result.stdout.decode().strip("\n")
+    if (not output.isnumeric()):
+        raise Exception("Result is not numeric")
+    return int(output)
+
+
+with (this_dir / 'fixtures' / 'commands.json').open() as file:
+    commands: Dict[str, List[Tuple[str, int]]] = json.load(file)
+
+testcases = []
+
+file: str
+for file in commands.keys():
```

Comment:
I believe that this should be equivalent, while preventing the `.keys()` list from being copied every iteration.
```suggestion
for file in commands:
```

Reply:
Updated in 877c3467b5fea8d60a39121c67e8f80a4e3c83a4.

I would add that:
1. The LHS operator in a python for...in loop is only evaluated once, and so the `keys()` function is only evaluated once when we enter the loop. ([docs](https://docs.python.org/3/reference/compound_stmts.html#for))
2. In python3, the `.keys` function (as well as `.values` and `.items`) was modified such that it returns a special view object that behaves like a generator and so the performance difference between `for file in commands` and `for file in commands.keys()` should be negligible. ([docs](https://docs.python.org/3/library/stdtypes.html#dictionary-view-objects))